### ISSUE TITLE
Enable File Based Message Storage

### ISSUE THREAD

**Issue by grugel-maintro** [SaraSmiseth/prosody#38]
Prosody allows "Internal" message storage. I currently use you docker image with my own prosody config files but i would like to switch to environment variables. Due to the way my backup instances keep the messages in sync it makes sense for me to enable the file based  [message storage](https://prosody.im/doc/storage) that can be enable with one line in the config file by adding the following line: `    storage = "internal"`. Is it possible to do that with an environment variable and if not can you implement this?

**Comment by SaraSmiseth** [comment]
Thanks for your suggestion.

I will add an environment variable for storage when I have time for testing and everything.

**Comment by SaraSmiseth** [comment]
Config looks like this now:

https://github.com/SaraSmiseth/prosody/blob/storage/conf.d/02-storage.cfg.lua

What about this part?

```lua
storage = {
  -- this makes mod_mam use the sql storage backend
  archive2 = os.getenv("STORAGE");
}
```

I guess I have to add variables for that as well. And make it optional.

**Comment by grugel-maintro** [comment]
Yes this is would be smart. I tested your commit but mod_mam wasn't working (so the messages weren't stored on the server as it seems. This is due to the use of  archive2 instead of just archive as the prosody mam man page tells:

> Legacy message archive
> 
> Early versions of mod_mam (which were available in [prosody-modules](https://prosody.im/community_modules)) stored data in a store called 'archive2'. This store is now just called 'archive'.
> 
> mod_mam can be instructed to use the older store name if you still have data there.
> 
> archive_store = "archive2"; -- the old data
> storage = {
>   archive2 = "sql";
> }

**Comment by SaraSmiseth** [comment]
Can you test the my updated PR? I renamed STORAGE to DEFAULT_STORAGE. And added a variable for ARCHIVE_STORE.

```yaml
DEFAULT_STORAGE: "internal"
ARCHIVE_STORE: "archive"
```
